tethysext/atcore/tests/integrated_tests/services/resource_workflows/helpers_tests.py

```diff
@@ -33,6 +33,20 @@ def test_set_step_status(self):
     @mock.patch('argparse._sys')
     def test_parse_workflow_step_args(self, mock_sys):
         mock_sys.argv = ['prog']
-        ret = helpers.parse_workflow_step_args()
+        ret, _ = helpers.parse_workflow_step_args()
 
         self.assertIsInstance(ret, Namespace)
+
+    @mock.patch('argparse._sys')
+    def test_parse_workflow_step_args_with_extra(self, mock_sys):
+        mock_sys.argv = ['prog']  # No extra argument
+        ret, extra_args = helpers.parse_workflow_step_args()
+
+        self.assertIsInstance(ret, Namespace)
+        self.assertListEqual(extra_args, [])
+
+        mock_sys.argv = ['prog', '--extra_arg']  # Extra argument
```

Comment:
This works but I'd prefer a test that is more similar to how it actually functions when it is used. There are usually a bunch of different arguments that are passed (sqlalchemy urls, class paths, geoserver urls, etc.) A better test would be to have argv populated with the number of values similar to those that would be passed in a real use case and additional args appended to the end.

Reply:
Test will be updated with the typical arguments (urls, id's, etc) followed by the extra arguments.